website/docs/reference/resource-properties/arguments.md

```diff
@@ -0,0 +1,117 @@
+---
+title: arguments
+sidebar_label: "arguments"
+id: arguments
+pagination_next: "reference/resource-properties/arguments"
+pagination_prev: null
+---
+
+<File name='macros/<filename>.yml'>
+
+```yml
+
+version: 2
+
+macros:
+  - name: <macro name>
+    arguments:
+      - name: <arg name>
+        type: <string>
+        description: <markdown_string>
+
+```
+
+</File>
+
+## Definition
+
+The `arguments` property is used to define the parameters that a macro can accept. Each argument can have a `name`, `type`, and `description`. This helps in documenting the macro and understanding what inputs it requires.
+
+You can use a [behavior flag](/reference/global-configs/behavior-changes#behavior-change-flags) in v1.10; dbt will validate whether the `name` or `type` matches the marcro arguments in YAML.
```

Comment:
I'm a little unclear on what this paragraph is communicating. We should probably mention which behavior flag you can use. Or are we saying that there is a behavior flag introduced in 1.10 that will affect macro arguments?